tests/e2e/e2e_test.go

```diff
@@ -1056,12 +1056,34 @@ func (s *IntegrationTestSuite) TestIBCWasmHooks() {
 	var response map[string]interface{}
 	s.Require().Eventually(func() bool {
 		response, err = nodeA.QueryWasmSmartObject(contractAddr, fmt.Sprintf(`{"get_total_funds": {"addr": "%s"}}`, senderBech32))
-		totalFunds := response["total_funds"].([]interface{})[0]
-		amount := totalFunds.(map[string]interface{})["amount"].(string)
-		denom := totalFunds.(map[string]interface{})["denom"].(string)
```

Comment:
I find the old versions much more readable here (since it's just a test). I'm not strongly against the type assertion checks and get the point in chain code, but in tests I tend to favor making the test shorted since robustness doesn't matter.

For json in particular, we may want to use https://github.com/tidwall/gjson